daemon/config/config_unix_test.go

```diff
@@ -30,8 +22,51 @@ func TestDaemonConfigurationMerge(t *testing.T) {
 			"log-opts": {
 				"tag": "test_tag"
 			}
-		}`))
+		}`)
+
+func TestGetConflictFreeConfiguration(t *testing.T) {
+	f, err := ioutil.TempFile("", "docker-config-")
+	if err != nil {
+		t.Fatal(err)
+	}
 
+	configFile := f.Name()
+
+	f.Write(configFileData)
+	f.Close()
+
+	config, err := getConflictFreeConfiguration(configFile, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !config.Debug {
+		t.Fatalf("expected %v, got %v\n", true, config.Debug)
+	}
+
+	if config.Ulimits == nil {
+		t.Fatal("expected default ulimit config, got nil\n")
+	}
+
+	if _, exist := config.Ulimits["nofile"]; !exist {
+		t.Fatal("expected default ulimit name nofile, got nil\n")
+	}
+
+	if config.Ulimits["nofile"].Name != "nofile" ||
+		config.Ulimits["nofile"].Hard != 2048 ||
+		config.Ulimits["nofile"].Soft != 1024 {
+		t.Fatalf("expected default ulimit name, hard and soft are nofile, 2048, 1024, got %s, %d, %d\n", config.Ulimits["nofile"].Name, config.Ulimits["nofile"].Hard, config.Ulimits["nofile"].Soft)
+	}
```

Comment:
All of this (from line 46) could be:

```go
expected := &units.Ulimit{...}
assert.Equal(t, expected, config.Ulimits)
```

Which will give you a nicely formatted error message when it fails.